Challenge2/src/agent.ts

```diff
@@ -13,25 +13,35 @@ import { UNI_SWAP_EVENT_ABI, UNI_POOL_FUNCTIONS_ABI, UNI_FACTORY_ADDRESS, UNI_IN
 
 const addressIsUniCache = new LRUCache<string, boolean>({ max: 100000 });
 
+export interface PoolValues {
+  token0: string;
+  token1: string;
+  fee: number;
+}
+
 const getPoolValues = async (
   provider: Provider,
   uniPoolFunctionsAbi: string[],
   poolAddress: string,
   txEvent: TransactionEvent
 ) => {
   const poolContract = new ethers.Contract(poolAddress, uniPoolFunctionsAbi, provider);
-  const poolValues: any[] = await Promise.all([
+  const [token0, token1, fee] = await Promise.all([
     poolContract.token0({ blockTag: txEvent.blockNumber }),
     poolContract.token1({ blockTag: txEvent.blockNumber }),
     poolContract.fee({ blockTag: txEvent.blockNumber }),
   ]);
-  return poolValues;
+  return { token0, token1, fee };
 };
 
-export const getRealPoolAddress = (uniFactoryAddress: string, uniInitCode: string, interceptedPoolValues: any[]) => {
+export const getRealPoolAddress = (
```

Comment:
Like I had mentioned previously, `getRealPoolAddress()` and `getPoolValues()` could be moved to a `utils.ts` file. Just to clean up this main `agent.ts` file. (Along with the comment about creating `types.ts`.)